crates/bindings/src/lib.rs

```diff
@@ -45,13 +42,610 @@ pub use timestamp::Timestamp;
+/// Any changes to the database will be rolled back.
+///
+/// Reducers are limited in their ability to interact with the outside world.
+/// They do not directly return data aside from errors, and have no access to any
+/// network or filesystem interfaces.
+/// Calling methods from [`std::io`], [`std::net`], or [`std::fs`]
+/// inside a reducer will result in runtime errors.
+///
+/// Reducers can communicate information to the outside world in two ways:
+/// - They can modify tables in the database.
+///   See the `#[table]`(#table) macro documentation for information on how to declare and use tables.
+/// - They can call logging macros from the [`log`] crate.
+///   This writes to a private debug log attached to the database.
+///   Run `spacetime logs <DATABASE_IDENTITY>` to browse these.
+///
+/// Reducers are permitted to call other reducers, simply by passing their `ReducerContext` as the first argument.
+/// This is a regular function call, and does not involve any network communication. The callee will run within the
+/// caller's transaction, and any changes made by the callee will be committed or rolled back with the caller.
+///
+/// # Lifecycle Reducers
+///
+/// You can specify special lifecycle reducers that are run at set points in
+/// the module's lifecycle. You can have one of each per module.
+///
+/// These reducers cannot be called manually
+/// and may not have any parameters except for `ReducerContext`.
+///
+/// ### `#[spacetimedb::reducer(init)]`
+///
+/// This reducer is run the first time a module is published
+/// and any time the database is cleared.
+///
+/// If an error occurs when initializing, the module will not be published.
+///
+/// ### `#[spacetimedb::reducer(client_connected)]`
+///
+/// This reducer is run when a client connects to the SpacetimeDB module.
+/// Their identity can be found in the sender value of the `ReducerContext`.
+///
+/// If an error occurs in the reducer, the client will be disconnected.
+///
+/// ### `#[spacetimedb::reducer(client_disconnected)]`
+///
+/// This reducer is run when a client disconnects from the SpacetimeDB module.
+/// Their identity can be found in the sender value of the `ReducerContext`.
+///
+/// If an error occurs in the disconnect reducer,
+/// the client is still recorded as disconnected.
+///
+/// ### `#[spacetimedb::reducer(update)]`
+///
+/// This reducer is run when the module is updated,
+/// i.e., when publishing a module for a database that has already been initialized.
+///
+/// If an error occurs when updating, the module will not be published,
+/// and the previous version of the module attached to the database will continue executing.
+///
+/// # Scheduled reducers
+///
+/// In addition to life cycle annotations, reducers can be made **scheduled**.
+/// This allows calling the reducers at a particular time, or in a loop.
+/// This can be used for game loops.
+///
```

Comment:
Should mention that scheduled reducers can still be called by clients, and suggest gating on `ctx.identity() == ctx.sender` to prevent that.